File: AlgorithmsProject/src/in/designpatterns/observer/ChangeDisplayElement.scala
package in.designpatterns.observer

class ChangeDisplayElement(observable: Observable) extends DisplayElement with Observer {
  var temperature: Float = _
  var humidity: Float = _
  var pressure: Float = _
  observable.addObserver(this)

  override def display =
    println("Current conditions: " + temperature
      + "F degrees and " + humidity + " % humidity");

  override def update(observable: Observable, objectPushed: Any) = {
    var weatherData: WeatherData = null
    weatherData = observable match {
      case n: WeatherData => observable.asInstanceOf[WeatherData]
    }
    if (weatherData != null) {
      temperature = weatherData.temperature
      humidity = weatherData.humidity
      pressure = weatherData.pressure
      display
    }
  }
}

File: AlgorithmsProject/src/in/designpatterns/scala/observer/ChangeDisplayElement.java
package in.designpatterns.scala.observer;

public class ChangeDisplayElement implements Observer, DisplayElement {
    private float temperature;
    private float humidity;
    private final Observable weatherData;

    public ChangeDisplayElement(Observable weatherData) {
        this.weatherData = weatherData;
        weatherData.registerObserver(this);
    }

    @Override
    public void update(float temp, float humidity, float pressure) {
        this.temperature = temp;
        this.humidity = humidity;
    }

    @Override
    public void display() {}

    public float getTemperature() { return temperature; }
    public float getHumidity() { return humidity; }
}
